### PR TITLE
Draft: octree memory

### PR DESCRIPTION
**Pull request by JStech**
This PR aims to update the memory usage in octree using modern C++ features.

### CODE REVIEW

File: octree/include/pcl/octree/impl/octree2buf_base.hpp [hunk 1]
```diff
@@ -60,7 +60,7 @@ Octree2BufBase<LeafContainerT, BranchContainerT>::~Octree2BufBase()
 {
   // deallocate tree structure
   deleteTree();
-  delete (root_node_);
+  root_node_.reset();
```
Comment (Morwenn) [inline octree/include/pcl/octree/impl/octree2buf_base.hpp:63]: This shouldn't be necessary I think: the memory will be automatically deleted by the `std::unique_ptr` destructor.

File: octree/include/pcl/octree/impl/octree_base.hpp [hunk 2]
```diff
@@ -62,7 +62,7 @@ OctreeBase<LeafContainerT, BranchContainerT>::~OctreeBase()
 {
   // deallocate tree structure
   deleteTree();
-  delete (root_node_);
+  root_node_.reset();
```
Comment (Morwenn) [inline octree/include/pcl/octree/impl/octree_base.hpp:65]: Same comment here.

**Comment by mvieth** [comment]
Is this still a draft? If yes, you could convert it to an actual draft-PR. If no, please change the title accordingly.

**Comment by JStech** [comment]
Definitely still a draft. Fixed.

**Pull request closed without merging**